github/commitstatus_test.go

```diff
@@ -192,23 +192,16 @@ OAuth: X-Accepted-Oauth-Scopes: , X-Oauth-Scopes: `,
 				MaxSleepTime: time.Minute,
 			}
 			ghStatus := github.NewCommitStatus(target, cfg.Token, cfg.Owner, cfg.Repo, context, hclog.NewNullLogger())
+
 			err := ghStatus.Add(cfg.SHA, state, targetURL, desc)
 
-			if err == nil {
-				t.Fatalf("\nhave: <no error>\nwant: %s", wantErr)
-			}
+			assert.Error(t, err, wantErr)
 			var ghError *github.StatusError
 			if !errors.As(err, &ghError) {
 				t.Fatalf("\nhave: %s\nwant: type github.StatusError", err)
 			}
 			wantStatus := tc.response.statuses[len(tc.response.statuses)-1]
-			if have, want := ghError.StatusCode, wantStatus; have != want {
-				t.Fatalf("status code: have: %d; want: %d", have, want)
-			}
-
-			if diff := cmp.Diff(wantErr, err.Error()); diff != "" {
```

Comment:
is this a mistake? why you don't want to check if the errors match? if you really want to remove that assert you can also cleanup all key values of `wantErr` in test cases.

Reply:
Thanks for looking closely at the code :-)

In this case, it should be correct, because

```go
assert.Error(t, err, wantErr)
```

replaces 2 asserts:
```go
if err == nil {
				t.Fatalf("\nhave: <no error>\nwant: %s", wantErr)
}

if diff := cmp.Diff(wantErr, err.Error()); diff != "" {
				t.Fatalf("error: (+have -want):\n%s", diff)
}
```

Reply:
Cool... Thanks